src/store/index.js

```diff
@@ -1,15 +1,56 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
+import { auth } from '../main';
 
 Vue.use(Vuex);
 
 export default new Vuex.Store({
   state: {
+    user: null,
+  },
+  getters: {
+    user: (state) => state.user,
   },
   mutations: {
+    setUser(state, payload) {
+      // this one is thorwing an error
+      //  state.user =[...payload]
+
+      // the following one will have issue if obj includes objs?
+      state.user = JSON.parse(JSON.stringify(payload));
+    },
   },
   actions: {
-  },
-  modules: {
+    async signupUser({ commit }, payload) {
+      try {
+        const cred = await auth.createUserWithEmailAndPassword(payload.email, payload.password);
+        console.log(cred);
+        commit('setUser', payload);
+      } catch (error) {
+        console.log(error);
+      }
+    },
+    async loginUser({ commit }, payload) {
+      try {
+        const cred = await auth.signInWithEmailAndPassword(payload.email, payload.password);
+        console.log(cred);
```

Comment:
please remove console.log, or use console.debug. The precompiler will remove the debug statements.

Reply:
done 
the change is as follow 
```
    async loginUser({ commit }, payload) {
      try {
        await auth.signInWithEmailAndPassword(payload.email, payload.password);
        commit('setUser', payload);
      } catch (error) {
        console.log(error);
      }
    },
```
I did the same process for singupUser as well.